src/firmware/app/control/trajectory_planner.h

```diff
@@ -11,293 +13,434 @@ typedef struct FirmwareRobotPathParameters
- * on the input parameters.
+ * Function balances the segment time durations of 2 separate speed profiles. The segment
+ * durations of profile1 and profile2 will be made equal - to the LONGEST duration at each
+ * point along the profiles. This is done by modifying the final speed elements.
  *
- * NOTE: This function does not break forwards continuity of the trajectory
+ * Note: Since displacement is the distance between speed nodes, there is 1 fewer element
+ * than the speed[] arrays. This is normal.
  *
- * @pre The input velocity_profile is required to be FORWARDS CONTINUOUS before it is used
- * as an input to this function
- * @pre forwards_continuous_velocity_profile is pre-allocated up to at least
- * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ * @pre The arrays must be pre-allocated up to at least
+ * TRAJECTORY_PLANNER_MAX_NUM_SEGMENTS.
+ *
+ * @param speeds1 [in] The speed profile corresponding to trajectory 1. Speed must have a
+ * denominator value of seconds.
  *
- * @param num_segments [in] The number of segments(elements) in the velocity_profile array
- * @param forwards_continuous_velocity_profile [in/out] The forwards continuous velocity
- * profile to ebe modified in-place into a profile that is also backwards continuous
- * @param arc_segment_length [in] The arc segment length of each segment in the path the
- * profile is being generated for in meters
- * @param max_allowable_acceleration [in] The max allowable acceleration at any point on
- * the velocity profile
+ * @param speeds2 [in] The speed profile corresponding to trajectory 2. Speed must have a
+ * denominator value of seconds.
+ *
+ * @param durations1 [in] The time durations corresponding to the length of time between
+ * successive speed points in trajectory 1. In seconds.
+ *
+ * @param durations2 [in] The time durations corresponding to the length of time between
+ * successive speed points in trajectory 2. In seconds.
+ *
+ * @param displacement1 [in] The displacement corresponding to the distance between
+ * successive speed elements in trajectory 1. In meters.
+ *
+ * @param displacement2 [in] The displacement corresponding to the distance between
+ * successive speed elements in trajectory 2. In meters.
+ *
+ * @param complete_time_profile [out] The complete time profile of the two balanced
+ * trajectories. This profile is absolute time to each point on the profile starting from
+ * zero. In seconds.
+ *
+ * @param num_elements [in] The number of elements in both trajectory 1 and 2.
+ *
+ * @return A status indicating whether or not generation was successful
  */
-void app_trajectory_planner_generateBackwardsContinuousVelocityProfile(
-    const unsigned int num_segments,
-    float forwards_continuous_velocity_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
-    const float arc_segment_length, const float max_allowable_acceleration);
+TrajectoryPlannerGenerationStatus_t app_trajectory_planner_modifySpeedsToMatchDuration(
```

Comment:
perhaps `app_trajectory_planner_modifySpeedsToMatchLongestDuration` might be more precise?

Reply:
Renamed